src/main/java/net/dv8tion/jda/api/utils/messages/AbstractMessageBuilder.java

```diff
@@ -0,0 +1,222 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.api.utils.messages;
+
+import net.dv8tion.jda.api.entities.IMentionable;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.interactions.components.LayoutComponent;
+import net.dv8tion.jda.internal.utils.AllowedMentionsImpl;
+import net.dv8tion.jda.internal.utils.Checks;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Abstract builder implementation of {@link MessageRequest}.
+ *
+ * <p>This builder cannot be instantiated directly. You should use {@link MessageCreateBuilder} or {@link MessageEditBuilder} instead.
+ *
+ * @param <T>
+ *        The result type used for {@link #build()}
+ * @param <R>
+ *        The return type used for method chaining
+ *
+ * @see   MessageCreateBuilder
+ * @see   MessageEditBuilder
+ */
+@SuppressWarnings({"ResultOfMethodCallIgnored", "unchecked"})
+public abstract class AbstractMessageBuilder<T, R extends AbstractMessageBuilder<T, R>> implements MessageRequest<R>
+{
+    protected final List<MessageEmbed> embeds = new ArrayList<>(Message.MAX_EMBED_COUNT);
+    protected final List<LayoutComponent> components = new ArrayList<>(Message.MAX_COMPONENT_COUNT);
+    protected final StringBuilder content = new StringBuilder(Message.MAX_CONTENT_LENGTH);
+    protected AllowedMentionsImpl allowedMentions = new AllowedMentionsImpl();
+
+    protected AbstractMessageBuilder() {}
+
+    @Nonnull
+    @Override
+    public R setContent(@Nullable String content)
+    {
+        if (content != null)
+        {
+            content = content.trim();
+            Checks.notLonger(content, Message.MAX_CONTENT_LENGTH, "Content");
+            this.content.setLength(0);
+            this.content.append(content);
+        }
+        else
+        {
+            this.content.setLength(0);
+        }
+        return (R) this;
+    }
+
+    @Nonnull
+    @Override
+    public String getContent()
+    {
+        return content.toString();
+    }
+
+    @Nonnull
+    @Override
+    public R mentionRepliedUser(boolean mention)
+    {
+        allowedMentions.mentionRepliedUser(mention);
+        return (R) this;
+    }
+
+    @Nonnull
+    @Override
+    public R allowedMentions(@Nullable Collection<Message.MentionType> allowedMentions)
```

Comment:
Should these mention methods be on the abstract one? Seems like they should only be on the Create builder as you can't retroactively mention ping someone in an edit

Reply:
You can edit the mentions on edit requests, you cannot add new pings, but you can affect whether it highlights the message.